src/Console/MigrateCommand.php

```diff
@@ -0,0 +1,103 @@
+
+        // check if we not in spanner mode
+        if(!$connection instanceof Connection) {
+            $command = 'migrate' . ($fresh ? ':fresh' : '');
+            $this->call($command, ['--seed' => $this->option('seed')]);
+            return;
+        }
+
+        // if running emulator, ensure instance exists
+        if (!empty(getenv('SPANNER_EMULATOR_HOST'))) {
+            $this->info('Checking Emulator Instance');
+            $spanner = new SpannerClient((array)$connection->getConfig('client'));
+            $instanceName = (string)$connection->getConfig('instance');
+            if (! $spanner->instance($instanceName)->exists()) {
+                $config = $spanner->instanceConfiguration('emulator-config');
+                $spanner->createInstance($config, $instanceName)->pollUntilComplete();
+                $this->info('Created Emulator Instance');
+            }
+        }
+
+        $this->info('Checking DB');
+        if(!$connection->databaseExists()) {
+            $this->info('Creating DB');
+            $connection->createDatabase();
+        }
+
+        if($fresh) {
+            $this->info('Dropping all tables');
+            $connection->getSchemaBuilder()->dropAllTables();
+        }
+
+        $this->info('Generating batch DDL');
+
+        /** @var Migrator */
+        $migrator = app('migrator');
+        $migrationPath = $this->laravel->databasePath().DIRECTORY_SEPARATOR.'migrations';
+        $migrations = $migrator->getMigrationFiles($migrationPath);
+        $repository = $migrator->getRepository();
+        $repositoryExists = $repository->repositoryExists();
+        $ran = $repositoryExists ? $repository->getRan() : [];
+        $getPending = Closure::bind(fn($files, $ran) => $this->pendingMigrations($files, $ran), $migrator, $migrator);
+        $migrations = $getPending($migrations, $ran);
+
+        foreach ($migrations as $path) {
+            $this->output->write(str_replace(base_path(), '', $path), true);
+        }
+
+        $ddl = [];
+        $dml = [];
+        $results = $connection->pretend(function() use ($migrator, $migrations, $repository, $repositoryExists) {
+            if(!$repositoryExists)
+                $repository->createRepository();
+            $migrator->runPending($migrations);
+        });
+
+        foreach($results as $result) {
+            $query = $result['query'];
+            if(Str::startsWith($query, 'select')) continue;
+
+            if(Str::startsWith($query, ['update', 'insert'])) 
+                $dml[] = $result;
+            else
+                $ddl[] = $query;
+        }
+
+        if(count($ddl)) {
+            $this->info('Running batch DDL');
+            $connection->runDdlBatch($ddl);
+        } else
+            $this->info('Nothing to migrate');
+
+        if(count($dml)) {
+            $this->info('Running DML');
+            foreach ($dml as $d) {
+                $connection->statement($d['query'], $d['bindings']);
```

Comment:
You can't just run DML's after, DDL's. In a case where DDL that runs after a DML might drop or alter the column you are trying to modify which will result in an error.

Reply:
This isnt an issue since `alter` and `drop` statements are DDL


https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_table_add_column_statement

https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#drop_table_statement

If you look at how `$dml` and `$ddl` is defined

```php
 foreach($results as $result) {
            $query = $result['query'];
            if(Str::startsWith($query, 'select')) continue;

            if(Str::startsWith($query, ['update', 'insert'])) 
                $dml[] = $result;
            else
                $ddl[] = $query;
        }
```

You will notice that I only add `insert` or `update` statements to `$dml` - any `alter` or `drop` statements will be run first, since they are added to `$ddl`

Reply:
From what I understand, the only `insert` statements that the migrations generate is inserts into the `migrations` table?

What other insert/update statements would a migration make? Migrations deal with defining tables/columns/indexes, not inserting data, that is for the seeders